FinSearchUnified/Tests/Components/ConfigLoaderTest.php

```diff
@@ -173,18 +195,18 @@ public function testWarmUpCache($responseCode, $responseBody, array $expectedCon
         $method->invoke($configLoader);
     }
 
-    public function warmUpCacheProvider()
+    public function cacheLoadProvider()
     {
         return [
-            'Config file is empty or null' => [200, null, []],
-            'Config file returns extra parameters' => [
-                200,
-                json_encode([
-                    'isStagingShop' => true,
-                    'directIntegration' => ['enabled' => false, 'iShouldNotBeHere' => true]
-                ]),
-                ['isStagingShop' => true, 'directIntegration' => ['enabled' => false]]
-            ],
+            'Cache load return false' => [false, $this->exactly(2)],
```

Comment:
Remove this data provider and split the test up in two cases. These are two separate scenarios, they are checking a different result so they should be separate tests.

Reply:
I am sorru but how is this a separate scenario? We are checking for the outcome of the cache load method, where the first scenario does not return the cache, whereas the second does return a cache.

Reply:
Yes so you basically have two different results. In the first scenario you want to ensure that `save` is called and in the other one you want to ensure that `save` is never called. Having an if inside of a test already tells you that it's probably a better idea to split it up.

A test should always test the same scenario and shouldn't do things based on what data the dataprovider returns. 

For example:

```php
public function someProvider()
{
    return array(
        array(0, 0, 0),
        array(0, 1, 1),
        array(1, 0, 1),
        array(1, 1, 3)
    );
}
```

Bad test case:

```php
/**
 * @dataProvider someProvider
 */
public function testAdd($a, $b, $c)
{
    if ($a = 0) {
        $this->assertNotEquals($c, $a + $b);
    } else {
        $this->assertEquals($c, $a + $b);
    }
}
```

Good test case:

```php
/**
 * @dataProvider someProvider
 */
public function testAdd($a, $b, $c)
{
    $this->assertEquals($c, $a + $b);
}
```

Of course this is an oversimplified example, but please split up the test if your assertions differ.

Reply:
@TheKeymaster OK I guess I understand the cases. But then this would require a lot of refactoring in other test cases as well because a lot of tests are written in this way. So maybe we can do this optimization and refactoring in a future story if possible.